static/js/src/modules/search.js

```diff
@@ -1,301 +1,402 @@
-const { el, mount, text, list, setChildren, setStyle, setAttr } = redom
+const { el, mount, text, list, setChildren, setStyle, setAttr } = redom;
 
-const projectList = ["Spin", "Cloud", "Bartholomew"]
+const projectList = ["Spin", "Cloud", "Bartholomew"];
 const versionPattern = /^v\d+$/;
-let idx
-let documents
+let idx;
+let documents;
 
 async function getSearchIndex() {
-    try {
-        let res = await fetch('/static/data.json')
-        return await res.json()
-    }
-    catch (err) {
-        console.log("cannot load search module")
-    }
+  try {
+    let res = await fetch("/static/data.json");
+    return await res.json();
+  } catch (err) {
+    console.log("cannot load search module");
+  }
 }
 
 async function setupSearch() {
-    documents = await getSearchIndex()
-    let currentPath = window.location.pathname
-    let splitPath = currentPath.split("/")
-    let version = splitPath[2]
-    if (version == "v1") {
-        documents = documents.filter(k => {
-            if (k.project != "spin") {
-                return true
-            }
-            return k.url.includes("spin/v1/")
-        })
-    } else {
-        documents = documents.filter(k => {
-            return k.project != "spin" || !k.url.includes("spin/v1/")
-        })
-    }
+  documents = await getSearchIndex();
+  let currentPath = window.location.pathname;
+  let splitPath = currentPath.split("/");
+  let version = splitPath[2];
+  if (version == "v1") {
+    documents = documents.filter((k) => {
+      if (k.project != "spin") {
+        return true;
+      }
+      return k.url.includes("spin/v1/");
+    });
+  } else if (version == "v2") {
+    documents = documents.filter((k) => {
+      if (k.project != "spin") {
+        return true;
+      }
+      return k.url.includes("spin/v2/");
+    });
+  } else if (version == "v3") {
+    documents = documents.filter((k) => {
+      if (k.project != "spin") {
+        return true;
+      }
+      return k.url.includes("spin/v3/");
+    });
+  } else {
+    documents = documents.filter((k) => {
+      return (
+        k.project != "spin" ||
+        !k.url.includes("spin/v1/") ||
+        !k.url.includes("spin/v2/") ||
+        !k.url.includes("spin/v3/")
+      );
```

Comment:
In other case where not in Spin docs, I think we want to include only v3 docs. 
```suggestion
    documents = documents.filter((k) => {
      return (
        k.project != "spin" ||
        k.url.includes("spin/v3/")
      );
```